### PR TITLE
[OV] Fixed se_block fused pattern

### PR DESCRIPTION
**Pull request by l-bat**
### Changes

* Fixed problem with matching se_block

### Reason for changes

* To improve performance

### Related tickets

* 105079

### Tests

* SeBlockModel synthetic model

### CODE REVIEW

**Comment by l-bat** [comment]
run openvino pre-commit tests

File: nncf/common/scopes.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def should_consider_scope(
     :return: A boolean value specifying whether a serializable_id should be considered (i.e. "not ignored", "targeted")
     """
     string_id = str(serializable_id)
-    return (target_scopes is None or matches_any(string_id, target_scopes)) \
+    return (not target_scopes or matches_any(string_id, target_scopes)) \
```
Comment (alexsu52) [inline nncf/common/scopes.py:73]: ?

File: nncf/common/quantization/quantizer_propagation/solver.py [hunk 2]
```diff
@@ -365,9 +369,16 @@ def __init__(self,
           and ignore matching nodes. Ignored nodes will not have quantizers applied to their inputs (even if
           required by node's metatype and HW config), and the downstream quantizers will not propagate upwards
           through the corresponding node.
+        :param ignored_scopes_per_group: A dict stores mapping of quantizer group to list of strings
+          to match against NNCFGraph node names and ignore matching nodes.
+          Ignored nodes will not have quantizers applied to corresponing quantizer group.
```
Comment (alexsu52) [inline nncf/common/quantization/quantizer_propagation/solver.py:374]: All quantizer groups should be present in the dict?

File: nncf/common/quantization/quantizer_propagation/solver.py [hunk 3]
```diff
@@ -443,15 +462,47 @@ def __init__(self,
...
+                    target_scopes_per_group[QuantizerGroup.WEIGHTS]
+                )
+            for x in quantizable_layer_nodes:
+                node_name = x.node.node_name
+                if should_consider_scope(node_name,
+                                        ignored_scopes=weights_ignored_scopes,
+                                        target_scopes=weights_target_scopes):
+                    weight_quantizable_node_names_vs_qconfigs[node_name] = x.qconfig_list
+                else:
+                    nncf_logger.info(f"Ignored adding weight quantizer for: {node_name}")
```
Comment (alexsu52) [inline nncf/common/quantization/quantizer_propagation/solver.py:503]:
```suggestion
                    nncf_logger.debug(f"Ignored adding weight quantizer for: {node_name}")
```

File: nncf/common/quantization/quantizer_propagation/solver.py [hunk 4]
```diff
@@ -443,15 +462,47 @@ def __init__(self,
...
+            return []
+        return [list_or_str] if isinstance(list_or_str, str) else list_or_str
+
+    def _filter_by_weight_ignored_target_scopes(self,
+                                                quantizable_layer_nodes: List[QuantizableWeightedLayerNode],
+                                                ignored_scopes_per_group: Dict[QuantizerGroup, List[str]],
+                                                target_scopes_per_group: Dict[QuantizerGroup, List[str]]
+    ) -> Dict[NNCFNodeName, List[QuantizerConfig]]:
+        weight_quantizable_node_names_vs_qconfigs = {}
+        if quantizable_layer_nodes is not None:
```
Comment (alexsu52) [inline nncf/common/quantization/quantizer_propagation/solver.py:484]: IMHO: This allows you to minimize the number of spaces on the left.
```suggestion
        if quantizable_layer_nodes is None:
            return {}
```

**Review by alexsu52 — COMMENTED** [review]
Thanks for the contribution! Good work!

File: nncf/common/scopes.py (hunk 1 above)
Comment (l-bat) [reply nncf/common/scopes.py:73]: because `target_scopes` can be an empty List

Comment (vshampor) [reply nncf/common/scopes.py:73]: If `target_scopes` is an empty list, I would expect that this should mean that there are no target scopes, e.g. everything in the model is _not_ targeted.

File: nncf/common/quantization/quantizer_propagation/solver.py [hunk 5]
```diff
@@ -346,7 +348,9 @@ class QuantizerPropagationSolver:
 
     def __init__(self,
                  ignored_scopes: List[str] = None,
+                 ignored_scopes_per_group: Dict[QuantizerGroup, List[str]] = None,
                  target_scopes: List[str] = None,
+                 target_scopes_per_group: Dict[QuantizerGroup, List[str]] = None,
```
Comment (vshampor) [inline nncf/common/quantization/quantizer_propagation/solver.py:353]: Either remove the `ignored_scopes` and `target_scopes` parameters in lieu of `_per_group` versions, or have explicit `weight_ignored_scopes`/`activation_ignored_scopes`/`weight_target_scopes`/`activation_target_scopes`.

File: nncf/torch/quantization/algo.py [hunk 6]
```diff
@@ -227,22 +226,6 @@ def _get_default_qconfig(self, constraints: QuantizationConstraints = None):
             qconfig = constraints.apply_constraints_to(qconfig)
         return qconfig
 
-    def _should_consider_scope_for_group(self, node_name: NNCFNodeName, group: QuantizerGroup) -> bool:
```
Comment (vshampor) [inline nncf/torch/quantization/algo.py:230]: This was a well-defined method before, do not inline it.

**Review by vshampor — CHANGES_REQUESTED** [review]

File: nncf/torch/quantization/algo.py (hunk 6 above)
Comment (l-bat) [reply nncf/torch/quantization/algo.py:230]: Looks like duplication of `should_consider_scope`

**Review by vshampor — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]

**Pull request merged**